src/lib/string_sign/string_sign.ml

```diff
@@ -94,15 +94,43 @@ let%test_module "Sign_string tests" =
       let signature = sign ~signature_kind keypair.private_key s in
       verify ~signature_kind signature keypair.public_key s
 
-    let%test "Sign with testnet, fail to verify with mainnet" =
+    let%test "Sign, verify with other networks" =
+      let s = "Sky is blue" in
+      let signature_kind = Mina_signature_kind.Other_network "Foo" in
+      let signature =
+        sign ~signature_kind:(Other_network "Foo") keypair.private_key s
+      in
+      verify ~signature_kind signature keypair.public_key s
+
+    let%test "Sign with testnet, fail to verify with mainnet or other networks"
+        =
       let open Mina_signature_kind in
       let s = "Some pills make you larger" in
       let signature = sign ~signature_kind:Testnet keypair.private_key s in
-      not (verify ~signature_kind:Mainnet signature keypair.public_key s)
+      (not (verify ~signature_kind:Mainnet signature keypair.public_key s))
+      && not
+           (verify ~signature_kind:(Other_network "Foo") signature
+              keypair.public_key s )
```

Comment:
For this and all the below tests, could you also add a check that they can be verified with the intended signature kind?